src/mistral_common/tokens/tokenizers/instruct.py

```diff
@@ -30,24 +30,31 @@
     TokenizedType,
     Tokenizer,
 )
-from mistral_common.tokens.tokenizers.multimodal import MultiModalEncoder
+from mistral_common.tokens.tokenizers.image import ImageEncoder
 
 
 class InstructTokenizerBase(
     InstructTokenizer, Generic[InstructRequestType, FIMRequestType, TokenizedType, AssistantMessageType]
 ):
     r"""Base instruct tokenizer."""
 
-    def __init__(self, tokenizer: Tokenizer, mm_encoder: Optional[MultiModalEncoder] = None):
+    def __init__(self, tokenizer: Tokenizer, image_encoder: Optional[ImageEncoder] = None):
         r"""Initialize the instruct tokenizer.
 
         Args:
             tokenizer: The tokenizer to use.
-            mm_encoder: The multi-modal encoder to use if any.
+            image_encoder: The image encoder to use if any.
         """
         self.tokenizer = tokenizer
-        self.mm_encoder = mm_encoder
-        super().__init__(tokenizer, mm_encoder)
+        self.image_encoder = image_encoder
+        super().__init__(tokenizer, image_encoder)
+
+    @property
+    def mm_encoder(self) -> Optional[ImageEncoder]:
+        # this funtion is deprecated, use image_encoder instead
+        # TODO(Patrick) - throw a deprecation warning once
+        # changes applied to vllm and transformers
```

Comment:
```suggestion
        # deprecated - only allowed for tokenizers <= v11
```

Reply:
Hmm no this is code change that's independent from tokenizer files -> this should be deprecated

Reply:
As discussed in Slack, you're right, resolving the convo !